### DISCUSSION TITLE
Can't deploy to Heroku

### DISCUSSION THREAD

**Discussion by kellyprogramsthings** [prisma/prisma#11704] [Q&A] [answered]
This is probably user error so I didn't put it in issues.

I'm getting table already exist errors when I try to deploy. Locally I just delete the database and start over if I have a problem, since Prisma is perfectly happy to make a new database. But deleting a database on Heroku is a lot more work.

Output from "prisma migrate reset --force"
```
Datasource "db": PostgreSQL database "", schema "public" at "ec2-50-19-254-63.compute-1.amazonaws.com:5432"

Applying migration `20220205042944_init`

Database reset successful

The following migration(s) have been applied:

migrations/
  └─ 20220205042944_init/
    └─ migration.sql

✔ Generated Prisma Client (3.8.1 | library) to ./node_modules/@prisma/client in 
2.74s

Running seed command `node prisma/seed.js` ...

🌱  The seed command has been executed.
```

Output from deploy log:
```
Prisma schema loaded from prisma/schema.prisma
       Datasource "db": PostgreSQL database "", schema "public" at "ec2-50-19-254-63.compute-1.amazonaws.com:5432"
       
       1 migration found in prisma/migrations
       
       Applying migration `20220208005420_init`
Error: P3018
A migration failed to apply. New migrations cannot be applied before the error is recovered from. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve
Migration name: 20220208005420_init
Database error code: 42P07
Database error:
ERROR: relation "tag_types" already exists
DbError { severity: "ERROR", parsed_severity: Some(Error), code: SqlState("42P07"), message: "relation \"tag_types\" already exists", detail: None, hint: None, position: None, where_: None, schema: None, table: None, column: None, datatype: None, constraint: None, file: Some("heap.c"), line: Some(1103), routine: Some("heap_create_with_catalog") }
-----> Build failed
```

I don't know how to get back to just a blank database with none of the tables. And I don't know why it's not recognizing that the tables are all already there and the migration has already been run.

I'm getting very frustrated with Prisma.

**Comment by janpio** [comment]
`reset` already does what you want I think: https://www.prisma.io/docs/reference/api-reference/command-reference#migrate-reset It empties your database and then recerates and seeds everything. Afterwards you do not need to deploy anything any more as the tables will already be present.

(But note the note about this being a command for development environments.)

**Reply by floelhoeffel** [reply]
Are you iterating on a db on heroku (e.g. active development) or is this a production db?

**Comment by kellyprogramsthings** [comment, marked as answer]
It worked this time. So yeah, user error. I would swear I did the exact same thing this time as last time. Regardless, it works now, and I appreciate both of you helping me. Sorry for using up your time.

(Posted from the wrong account the first time.)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
